server/app/rest_api/rest.py

```diff
@@ -221,7 +221,7 @@ def is_safe_collection_name(name):
     """
     if name is None:
         return False
-    return re.match(r"^\w+$", name) is not None
+    return re.match(r"^[\w\-]+$", name) is not None
```

Comment:
```suggestion
    return re.match(r"^(?!\-)[\w\-]+(?<!\-)$", name) is not None
```

We should probably guard against the name starting or ending in a hyphen, right? Fix for this included.

Reply:
Why do we need to include this additional restriction?    -foobar is an entirely legal filename on any system I can think of.

Reply:
Fair enough. Resolving.